src/db_api_util.ts

```diff
@@ -1031,6 +1052,67 @@ export async function getThreadNeedsAttentionFor(
   }
 }
 
+// Return a list of ThreadInfo's for all threads needing followup
+export async function getThreadsNeedingFollowUp(
+  slackUserId: string,
+  days: number
+): Promise<ThreadInfo[]> {
+  const client = await pool.connect();
+  try {
+    const result = await client.query(
+      `WITH all_claims AS (
+        SELECT
+          user_id
+          , volunteer_slack_user_id
+          , volunteer_slack_user_name
+          , row_number () OVER (PARTITION BY user_id ORDER BY created_at DESC) AS rn
+        FROM volunteer_voter_claims
+        WHERE archived != true
+      ), latest_claims AS (
+        SELECT user_id, volunteer_slack_user_id, volunteer_slack_user_name
+        FROM all_claims
+        WHERE rn = 1
+      ), all_status AS (
+        SELECT
+          user_id, voter_status
+          , row_number() OVER (PARTITION BY user_id ORDER BY created_at DESC) AS rn
+        FROM voter_status_updates
+      ), latest_status AS (
+        SELECT user_id, voter_status
+        FROM all_status
+        WHERE rn = 1
+      )
+      SELECT
+        t.slack_parent_message_ts
+        , t.slack_channel_id
+        , t.user_id
+        , EXTRACT(EPOCH FROM now() - updated_at) as last_update_age
+        , c.volunteer_slack_user_name
+        , s.voter_status
+        FROM threads t, latest_claims c, latest_status s
+        WHERE
+          needs_attention = false
+          AND routed = false
+          AND updated_at <= NOW() - interval '${days} days'
+          AND t.user_id=c.user_id
+          AND c.volunteer_slack_user_id=$1
+          AND s.user_id = t.user_id
```

Comment:
don't you have to match twilio_phone_number = twilio_phone_number as well?

Reply:
oh right!  using is_demo instead (the phone # isn't in the threads table but is_demo is everywhere)

Reply:
can we add twilio_phone_number instead of is_demo to the threads table (or both)? twilio_phone_number and user_id are used throughout the code and in Redis for identifying a unique user, and I'd like to keep that consistent.